src/NuGet.Core/NuGet.ProjectModel/PackageSpecReferenceDependencyProvider.cs

```diff
@@ -393,11 +394,26 @@ internal List<LibraryDependency> GetSpecDependencies(
                     // must be external projects.
                     var dependency = dependencies[i];
                     var libraryRange = new LibraryRange(dependency.LibraryRange) { TypeConstraint = dependency.LibraryRange.TypeConstraint & ~LibraryDependencyTarget.Project };
-                    dependencies[i] = new LibraryDependency(dependency) { LibraryRange = libraryRange };
+                    dependencies[i] = new LibraryDependency(dependency)
+                    {
+                        LibraryRange = libraryRange,
+                        SuppressParent = isPruned ? LibraryIncludeFlags.All : dependency.SuppressParent,
+                        IncludeType = isPruned ? LibraryIncludeFlags.None : dependency.IncludeType,
+                    };
                 }
             }
 
             return dependencies;
+
+            static bool IsDependencyPruned(LibraryDependency dependency, IReadOnlyDictionary<string, PrunePackageReference> packagesToPrune)
+            {
+                if (packagesToPrune?.TryGetValue(dependency.Name, out PrunePackageReference packageToPrune) == true
+                    && dependency.LibraryRange.VersionRange.Satisfies(packageToPrune.VersionRange.MaxVersion))
+                {
+                    return true;
+                }
+                return false;
+            }
```

Comment:
It looks like this code is a duplicate of `IncludeFlagUtils.IsDependencyPruned`. This indicates to me that we could benefit from a data structure that handles the pruned dependency case and reduce the if conditionals. In lieu of that, we should deduplicate the code.

Reply:
It is duplicated, but there's no type that would naturally own this, since TargetFrameworkInformation itself is a model and it has a dict of prune package refs. 

The other consideration why I didn't add a type is that the feature is in development, so I didn't have enough scenarios to come up with an API. 

Maybe I do know, but as of right now I've landed in the camp of I'd rather have 2 copies than a bad API.